From 87c1c4ad987a526a19314dd0da043391ca8ee459 Mon Sep 17 00:00:00 2001
From: Daniel Krol <orblivion@gmail.com>
Date: Thu, 28 Oct 2021 21:53:23 -0400
Subject: [PATCH 13/13] Sandstorm comment permission into clientVars

(TODO - squash me into the readonly one thing I think. And/or actually, set the clientvars for canComment properly within that plugin.)
---
 src/node/handler/PadMessageHandler.js | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/node/handler/PadMessageHandler.js b/src/node/handler/PadMessageHandler.js
index 1693fc93..83da89ec 100644
--- a/src/node/handler/PadMessageHandler.js
+++ b/src/node/handler/PadMessageHandler.js
@@ -987,8 +987,10 @@ const handleClientReady = async (socket, message, authorID) => {
   sessionInfo.readOnlyPadId = padIds.readOnlyPadId;
 
   // SANDSTORM EDIT
-  var hasSandstormPermissions = decodeURIComponent(socket.client.request.headers["x-sandstorm-permissions"])
-      .split(",");
+  var hasSandstormPermissions = process.env.SANDSTORM
+      ? decodeURIComponent(socket.client.request.headers["x-sandstorm-permissions"]).split(",")
+      : ["modify", "comment"]; // when not in sandbox, pretend we have all sandstorm permissions
+
   var sandstormReadonly = hasSandstormPermissions.indexOf("modify") === -1;
   // Not sure what webaccess.userCanModify is, but it seems to work fine, and at worst it will
   // err on being too restrictive, so I'll keep it.
@@ -1122,6 +1124,7 @@ const handleClientReady = async (socket, message, authorID) => {
       numConnectedUsers: roomSockets.length,
       readOnlyId: padIds.readOnlyPadId,
       readonly: sessionInfo.readonly, // SANDSTORM EDIT - as etherpad changes, make sure this continues to be something that we set via x-sandstorm-permissions header
+      sandstormPermissions: hasSandstormPermissions, // SANDSTORM EDIT
       serverTimestamp: Date.now(),
       userId: authorID,
       abiwordAvailable: settings.abiwordAvailable(),
-- 
2.20.1

